util.go

```diff
@@ -18,13 +21,53 @@ func hash64(x uintptr) uint32 {
 	return uint32(x)
 }
 
-// FNV-1a 32-bit hash
-func fnv32(key string) uint32 {
-	hash := fnv32Offset
-	keyLength := len(key)
-	for i := 0; i < keyLength; i++ {
-		hash ^= uint32(key[i])
-		hash *= fnv32Prime
+//go:noescape
+//go:linkname memhash runtime.memhash
+func memhash(p unsafe.Pointer, h, s uintptr) uintptr
+
+func memhashMap(msg interface{}) uint64 {
+	switch val := msg.(type) {
+	case string:
+		strhv := *(*reflect.StringHeader)(unsafe.Pointer(&val))
+
+		return uint64(memhash(unsafe.Pointer(strhv.Data), 0, uintptr(strhv.Len)))
+	case []byte:
```

Comment:
[]byte slices aren't comparable (see [the spec](https://golang.org/ref/spec#Comparison_operators)) which is problematic because we need to check keys for equality.

That could be worked around by applying custom logic for []byte slice equality checks, but such check should also check types of both keys.